hw/top_darjeeling/ip_autogen/racl_ctrl/doc/programmers_guide.md

```diff
@@ -0,0 +1,27 @@
+# Programmer's Guide
+
+RACL Control can be configured through its [registers](./registers.md).
+
+## RACL Policies
+
+Each RACL policy has a register named after the policy which contains a permission bitmap for read and write permissions for each of the roles defined in the top-level configuration.
+The exact register layout is defined in [RACL: Register Access Control List](../../../../../doc/contributing/hw/racl/README.md#racl-policies).
+Once set, the policies (i.e. permission bitmaps) are distributed to all subscribing IPs and used for subsequent RACL access checks.
+
+## Interrupts
+
+When a RACL error occurs, an interrupt is raised.
+Interrupts are disabled by default but can be enabled via the [`INTR_ENABLE`](./registers.md#intr_enable) register.
+
+## Error Logs
+
+When a RACL error occurs, the following information about the error is logged in the [`ERROR_LOG`](./registers.md#error_log) and [`ERROR_LOG_ADDRESS`](./registers.md#error_log_address) registers.
```

Comment:
Do we need to give programmers any guidance on how the registers should be read and written? I'm thinking of something along the lines of
- Wait for a RACL error interrupt or poll the `ERROR_LOG.valid` bit.
- Read the `ERROR_LOG_ADDRESS` register.
- Read the `ERROR_LOG` register.
- Clear the log entry by writing `1` to the `ERROR_LOG.valid` bit.

It may seem obvious but could help avoid questions or usage bugs.